eclipse-collections-code-generator/src/main/resources/impl/set/mutable/primitiveHashSet.stg

```diff
@@ -1615,7 +1615,21 @@ public class <name>HashSet extends Abstract<name>Set implements Mutable<name>Set
             {
                 removeValue = <name>HashSet.this.table[this.position - 1];
             }
-            <name>HashSet.this.remove(removeValue, false);
+            if (<name>HashSet.isBetweenZeroAndThirtyOne(removeValue))
+            {
+                <name>HashSet.this.removeZeroToThirtyOne(removeValue);
+            }
+            else if (<(equals.(type))({<name>HashSet.this.table[this.position - 1]}, "removeValue")>)
+            {
+                if (<name>HashSet.this.copyOnWrite)
+                {
+                    <name>HashSet.this.copyTable();
```

Comment:
@motlin @gs-rezaem this was copied from the `remove()` implementation, but I do not think this if statement is necessary. `this.copyOnWrite` will always be false inside the iterator. I tried writing a test case which will get it to be true but did not find a condition. May be I am missing something. Happy to remove the if statement or add the test case which will enter the branch.